test/connectors/httpParsers/CalculationHttpParserSpec.scala

```diff
@@ -0,0 +1,251 @@
+
+      "return a Error instance" in {
+        result shouldEqual expected
+      }
+    }
+
+    "the http response status is 400 BAD_REQUEST (multiple errors)" should {
+
+      val httpResponse: AnyRef with HttpResponse = HttpResponse(Status.BAD_REQUEST,
+        responseJson = Some(Json.obj(
+          "failures" -> Json.arr(
+            Json.obj(
+              "code" -> "ERROR CODE 1",
+              "reason" -> "ERROR MESSAGE 1"
+            ),
+            Json.obj(
+              "code" -> "ERROR CODE 2",
+              "reason" -> "ERROR MESSAGE 2"
+            )
+          )
+        ))
+      )
+
+      val expected: Either[ErrorResponse, Nothing] = Left(ErrorResponse(
+        Status.BAD_REQUEST,
+        MultiError(
+          failures = Seq(
+            Error(code = "ERROR CODE 1", reason = "ERROR MESSAGE 1"),
+            Error(code = "ERROR CODE 2", reason = "ERROR MESSAGE 2")
+          )
+        )
+      ))
+
+      val result: CalculationHttpParser.HttpGetResult[PreviousCalculationModel] = PreviousCalculationReads.read("", "", httpResponse)
+
+      "return a MultiError" in {
+        result shouldEqual expected
+      }
+
+    }
+
+    "the http response status is 400 BAD_REQUEST (Unexpected Json Returned)" should {
+
+      val httpResponse: AnyRef with HttpResponse = HttpResponse(Status.BAD_REQUEST, responseJson = Some(Json.obj("foo" -> "bar")))
+
+      val expected: Either[UnexpectedJsonFormat.type, Nothing] = Left(UnexpectedJsonFormat)
+
+      val result: CalculationHttpParser.HttpGetResult[PreviousCalculationModel] = PreviousCalculationReads.read("", "", httpResponse)
+
+      "return an UnexpectedJsonFormat instance" in {
+        result shouldEqual expected
+      }
+
+    }
+
+    "the http response status is 400 BAD_REQUEST (Bad Json Returned)" should {
+
+      val httpResponse: AnyRef with HttpResponse = HttpResponse(Status.BAD_REQUEST, responseString = Some("Banana"))
+
+      val expected: Either[InvalidJsonResponse.type, Nothing] = Left(InvalidJsonResponse)
+
+      val result: CalculationHttpParser.HttpGetResult[PreviousCalculationModel] = PreviousCalculationReads.read("", "", httpResponse)
+
+      "return an UnexpectedJsonFormat instance" in {
+        result shouldEqual expected
+      }
+
+    }
+
+    "the http response status is 500 Internal Server Error" should {
+
+      val httpResponse: AnyRef with HttpResponse = HttpResponse(Status.INTERNAL_SERVER_ERROR,
+        responseJson = Some(Json.obj(
+          "code" -> "code",
+          "reason" -> "reason"
+        ))
+      )
+
+      val expected: Either[ErrorResponse, Nothing] = Left(ErrorResponse(
+        Status.INTERNAL_SERVER_ERROR,
+        Error(
+          code = "code",
+          reason = "reason"
+        )
+      ))
+
+      val result: CalculationHttpParser.HttpGetResult[PreviousCalculationModel] = PreviousCalculationReads.read("", "", httpResponse)
+
+      "return an Internal Server Error" in {
```

Comment:
It might be worth changing these last two test names to follow suit on the tests above where they are mentioning the specific type of custom error returned as opposed to the status within the error.